lib/plug/cowboy/handler.ex

```diff
@@ -15,13 +22,24 @@ defmodule Plug.Cowboy.Handler do
       {:ok, req, {plug, opts}}
     catch
       kind, reason ->
-        exit_on_error(kind, reason, System.stacktrace(), {plug, :call, [conn, opts]})
+        stacktrace = System.stacktrace()
+        metadata = %{kind: kind, error: reason, stacktrace: stacktrace, name: :plug_cowboy}
+        measurements = %{duration: System.monotonic_time() - start}
+        :telemetry.execute([:plug_adapter, :request, :exception], measurements, metadata)
+
+        exit_on_error(kind, reason, stacktrace, {plug, :call, [conn, opts]})
     after
       receive do
         @already_sent -> :ok
       after
-        0 -> :ok
+        0 ->
+          :ok
       end
+    else
+      result ->
+        measurements = %{duration: System.monotonic_time() - start}
+        :telemetry.execute([:plug_adapter, :request, :stop], measurements, %{name: :plug_cowboy})
```

Comment:
Ideally we'd have the updated `conn` in this event... This is how I did the `:stop` event in my PR:

https://github.com/elixir-plug/plug_cowboy/pull/27/files#diff-497f9396e978da4bf4bfa189df9f6f13R19-R29

Reply:
Did you see the discussion at https://github.com/elixir-plug/plug_cowboy/pull/25#discussion_r406338510? I'm happy to add it back.

Reply:
I mean, w/o the `conn` I'd say this instrumentation is pretty pointless. There'd be no way to aggregate on anything, so we'd just literally get measurements like average request time for every request.

Reply:
Let's add the connection to all events then (and in this case it is `result`). I also propose to rename `name: :plug_cowboy` to `adapter: :plug_cowboy`. Thoughts?

Reply:
Is the updated `conn` part of the `result`? It looks like only `req` is extracted & returned

Reply:
A problem I ran into in my PR is that there's no way that I could figure out to get ahold of an updated `conn` in the case of an exception being raised.

Reply:
Let’s return the conn then from try. There is nothing we can do about catch though, it is a non local return.

Reply:
This is the issue I ran into with the Tesla PR where you may want to know something about the conn in an exception. If you reference it from there it is the original argument.

> Let’s return the conn then from try.

What is the suggestion here? I'm a little confused.

Reply:
I think it's to assign the result of `maybe_send` inside the `try` block & call `:telemetry.execute` there...

https://github.com/elixir-plug/plug_cowboy/pull/27/files#diff-497f9396e978da4bf4bfa189df9f6f13R19-R28